src/ui/common/src/components/integrations/dialogs/snowflakeDialog.tsx

```diff
@@ -113,6 +114,26 @@ export const SnowflakeDialog: React.FC<Props> = ({
         onChange={(event) => onUpdateField('password', event.target.value)}
         value={value?.password ?? null}
       />
+
+      <IntegrationTextInputField
+        spellCheck={false}
+        required={false}
+        label="Role"
+        description="The role to use when accessing the database above."
+        placeholder={Placeholders.role}
+        onChange={(event) => onUpdateField('role', event.target.value)}
+        value={value?.role ?? null}
+      />
     </Box>
   );
 };
+
+export function isSnowflakeConfigComplete(config: SnowflakeConfig): boolean {
```

Comment:
This doesn't validate the form correctly unfortunately. See this screenshot, I'm able to get a blue confirm button without entering a username field:

<img width="1264" alt="image" src="https://user-images.githubusercontent.com/1031759/222266090-5c0da95d-cde1-40bd-b2c8-f07bf6ba6cf4.png">

Instead, what I might recommend here to get this PR out the door is to check the length of each required input and make sure that it's > 0.

In the future, we should look into using a library like react-hooks-form or rolling our own Form component to do validation.

Reply:
I could even trigger it like this:
<img width="1263" alt="image" src="https://user-images.githubusercontent.com/1031759/222267037-72672b5e-7406-4022-9461-01af9f44734f.png">

Reply:
Fixed